lib/controller.js

```diff
@@ -0,0 +1,325 @@
+        region += (refSeparator + start + '-' + end);
+      } else {
+        if (start !== 1) {
+          region += refSeparator + start;
+        }
+      }
+      url += ('&region=' + region);
+    } else {
+      if ((typeof query.start !== 'undefined') || (typeof query.end !== 'undefined')) {
+        throw new Error(
+          "'referenceName' attribute requered if 'start' or 'end' attribute is given");
+      }
+    }
+
+    return url;
+  }
+
+  _redirect(id, query) {
+    try {
+      let url = this._ga4ghToDirectUrl(id, query);
+      let reply = {};
+      reply.urls = [url];
+      let redirect = JSON.stringify(reply);
+      this.response.removeHeader('Transfer-Encoding');
+      this.response.writeHead(200, 'OK, see redirection instructions in the body of the message', {
+        'Content-Length': redirect.length,
+        'Content-Type':   'application/json'});
+      this.response.write(redirect);
+    } catch (err) {
+      this.errorResponse(this.response, 422, err);
+    } finally {
+      this.response.end();
+    }
+  }
+
+  /**
+   * Interprets a query url and its parameters.
+   * Maps the query to data sources and authorises the client for access.
+   * Handles request and sends a reply to the client.
+   *
+   * User authentication should be performed elsewhere. It is expected that the
+   * 'x-remote-user' header is set in the request.
+   *
+   * User authorisation is per data resource, see 'auth' module for details.
+   *
+   * @param host - the name of the host to retrieve the data from.
+   *               If the data are not available on the given host, the
+   *               default host will be used.
+   *
+   * The following URLs are accepted:
+   *   URL           Paramaters
+   *   /file         name      - required
+   *                 directory - optional
+   *                 format    - required data format; 'sam', 'cram' and 'bam'
+   *                             are recognised; if the format is not one of
+   *                             these three or not given, 'bam' is used
+   *                 region    - optional, in the format 'chr1:3000-3020', multiple
+   *                             key-value pairs are allowed
+   *   /sample       accession - sample accession number, required
+   *                 region    - as for /file, but a single key-value pair only
+   *                 format    - as for /file
+   *   /api/ga4gh/v.0.1/get/sample/XXXX, where XXXX is a sample accession number
+   *                 format    - as above
+   *                 referenceName - chromosome name, optional
+   *                 start     - The start position of the range on the reference,
+   *                             0-based inclusive; optional, if specified, referenceName
+   *                             must also be specified
+   *                 end       - The end position of the range on the reference,
+   *                             0-based inclusive; optional, if specified, referenceName
```

Comment:
On 03/06/2016 15:02, dkj wrote:
>
> exclusive for end - no?
>
>
Thank you, fixed.




-- 
 The Wellcome Trust Sanger Institute is operated by Genome Research 
 Limited, a charity registered in England with number 1021457 and a 
 company registered in England with number 2742969, whose registered 
 office is 215 Euston Road, London, NW1 2BE.